### PR TITLE
Issue 2497

### PR DESCRIPTION
**Pull request by maplecomdori**
The pop up modal for adding annotations now gets an autofocus in the first textArea or input tag. 
Changes were made in the applications.js to delay the focus after the modal is finished loading by waiting 0.2 second. 
I tested with the Markus instance in my local machine.

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -14,9 +14,22 @@
...
+
 function ModalMarkus(elem) {
   this.modal_dialog = jQuery(elem).easyModal({
-    updateZIndexOnOpen: false
+      onOpen: function(myModal){
+          setTimeout(function() {
+              // wait for the modal to load
+
+              // search for elements that can receive text as input
+              var input = jQuery(myModal).find("textArea, input:text");
```
Comment (david-yz-liu) [inline app/assets/javascripts/application.js]: Change "textArea" to "textarea"

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -14,9 +14,22 @@
...
+              if(input[0] != null) {
```
Comment (david-yz-liu) [inline app/assets/javascripts/application.js]: put a space after the `if`, and check `input.length > 0`

File: app/assets/javascripts/application.js (hunk 1 above)
Comment (david-yz-liu) [inline app/assets/javascripts/application.js]: Also please change the variable name to `inputs`

File: app/assets/javascripts/application.js [hunk 3]
```diff
@@ -14,9 +14,22 @@
 
 /** Modal windows, powered by jQuery.easyModal. */
 
+
 function ModalMarkus(elem) {
   this.modal_dialog = jQuery(elem).easyModal({
-    updateZIndexOnOpen: false
+      onOpen: function(myModal){
+          setTimeout(function() {
+              // wait for the modal to load
```
Comment (david-yz-liu) [inline app/assets/javascripts/application.js]: This is a good comment, but should go before the `setTimeout` line

File: app/assets/javascripts/application.js [hunk 4]
```diff
@@ -14,9 +14,22 @@
 
 /** Modal windows, powered by jQuery.easyModal. */
 
+
 function ModalMarkus(elem) {
   this.modal_dialog = jQuery(elem).easyModal({
-    updateZIndexOnOpen: false
+      onOpen: function(myModal){
```
Comment (david-yz-liu) [inline app/assets/javascripts/application.js]: put a space before the `{`

File: app/assets/javascripts/application.js [hunk 5]
```diff
@@ -14,9 +14,22 @@
...
+          setTimeout(function() {
+              // wait for the modal to load
+
+              // search for elements that can receive text as input
+              var input = jQuery(myModal).find("textArea, input:text");
+              if(input[0] != null) {
+                  input[0].focus();
+              }
+
```
Comment (david-yz-liu) [inline app/assets/javascripts/application.js]: Please remove the blank lines here and below

File: app/assets/javascripts/application.js [hunk 6]
```diff
@@ -14,9 +14,22 @@
 
 /** Modal windows, powered by jQuery.easyModal. */
 
+
```
Comment (david-yz-liu) [inline app/assets/javascripts/application.js]: Delete this blank line

File: app/assets/javascripts/application.js [hunk 7]
```diff
@@ -16,7 +16,17 @@
 
 function ModalMarkus(elem) {
   this.modal_dialog = jQuery(elem).easyModal({
-    updateZIndexOnOpen: false
+      onOpen: function(myModal) {
+          // wait for the modal to load
+          setTimeout(function() {
+              // search for elements that can receive text as input
+              var inputs = jQuery(myModal).find("textarea, input:text");
+              if (inputs[0] != null && inputs.length > 0) {
```
Comment (david-yz-liu) [inline app/assets/javascripts/application.js]: @maplecomdori Checking inputs.length > 0 should be enough; you can remove the `inputs[0] != null` check.

**Comment by maplecomdori** [comment]
removed the `inputs[0] != null` check.

**Comment by david-yz-liu** [comment]
@maplecomdori I don't think you committed your last change (to remove that check).

**Comment by david-yz-liu** [comment]
Thanks, @maplecomdori. Good work.

**Pull request merged**